google_guest_agent/non_windows_accounts_integ_test.go

```diff
@@ -1,12 +1,76 @@
+//go:build integration
 // +build integration
 
 package main
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
+	"strings"
 	"testing"
 )
 
+const (
+	testUser       = "integration-test-user"
+	expectedGroups = []string{"adm", "dip", "docker", "lxd", "plugdev", "video", "google-sudoers"}
+)
+
+func TestCreateGoogleUser(t *testing.T) {
+	if exist, err := userExists(testUser); exist == true {
+		t.Errorf("test user should not exist")
+	}
+	if err := createGoogleUser(testUser); err != nil {
+		t.Errorf("failed creating test user")
+	}
+	if exist, err := userExists(testUser); exist != true || err != nil {
+		t.Errorf("tesr user should exist")
+	}
+	cmd := exec.Command("groups", testUser)
+	ret := runCmdOutput(cmd)
+	if ret.ExitCode() != 0 {
+		t.Errorf("test user should be added to group")
+	}
+	groups := strings.Split(strings.TrimSpace(strings.Split(ret.Stdout(), ":")[1]), " ")
+	for _, group := range groups {
+		if !contains(group, expectedGroups) {
+			t.Errorf("test user has been added to an unexpected group %s", group)
+		}
+	}
+	for _, expected := range expectedGroups {
+		if !contains(expected, groups) {
+			t.Errorf("test user has not been added to group %s", expected)
+		}
+	}
```

Comment:
think we should not do this check. the list of supplementary groups is what we will *try* to add the user to, but the group itself may not exist. we don't guarantee that every group listed in `[Accounts].groups` will be added because we don't create the groups.
```suggestion
```